Source/Core/Core.js

```diff
@@ -368,6 +368,7 @@ var cloneOf = function(item){
 };
 
 Array.implement('clone', function(){
+	if(this.slice) return this.slice(0);
```

Comment:
`Array.clone` clones also eventual objects inside the array. Doing a early return here using `return this.slice(0);` would keep the objects reference and add a new behaviour.